math/matrix/addmatrices.go

```diff
@@ -0,0 +1,27 @@
+package matrix
+
+import (
+	"errors"
+
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+type Number interface {
```

Comment:
Why make a new interface? You can directly use `constraints.Integer` in the source.